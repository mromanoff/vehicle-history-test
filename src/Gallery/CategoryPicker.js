import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ds } from '../designSystem';

const Button = styled.button`
  font-size: ${ds.fontSize('s')};
  cursor: pointer;
  outline: none;
  background: ${props =>
    props.active ? ds.color('secondary') : props.theme.color.secondary3};
  color: ${props => props.theme.color.white};
  transition: background-color 100ms ease-out 100ms;
  padding: ${props => props.theme.size.xs} ${props => props.theme.size.sm};

  &:hover,
  &.active {
    background: ${props => props.theme.color.secondary4};
  }
`;

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 2rem;
`;

class CategoryPicker extends Component {
  static propTypes = {
    categories: PropTypes.array.isRequired,
    currentCategory: PropTypes.string.isRequired,
    onCategoryChange: PropTypes.func.isRequired,
  };

  state = {};

  onClick(category) {
    this.props.onCategoryChange(category);
    this.setState({ category: category });
  }

  componentDidMount() {
    const { onCategoryChange, currentCategory } = this.props;
    if (currentCategory) {
      this.setState({ category: currentCategory });
      onCategoryChange(currentCategory);
    }
  }

  render() {
    return (
      <section>
        <Wrapper>
          {this.props.categories.map(category => {
            return (
              <Button
                key={category}
                active={this.state.category === category}
                onClick={() => this.onClick(category)}
              >
                {category}
              </Button>
            );
          })}
        </Wrapper>
      </section>
    );
  }
}

export default CategoryPicker;
