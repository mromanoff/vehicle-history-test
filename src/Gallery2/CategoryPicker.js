import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../components/Button';

class CategoryPicker extends Component {
  static propTypes = {
    categories: PropTypes.array.isRequired,
    initialCategory: PropTypes.string.isRequired,
    onCategoryPick: PropTypes.func.isRequired,
  };

  state = {};

  onClick(category) {
    this.props.onCategoryPick(category);
    this.setState({ category: category });
  }

  componentDidMount() {
    const { onCategoryPick, initialCategory } = this.props;
    if (initialCategory) {
      this.setState({ category: initialCategory });
      onCategoryPick(initialCategory);
    }
  }

  render() {
    return (
      <section>
        <h1 className="title">Pick category</h1>
        <div className="module category-picker">
          {this.props.categories.map(category => {
            return (
              <Button
                key={category}
                className={`btn btn--tertiary btn--sm ${
                  this.state.category === category ? 'btn--tertiary-active' : ''
                }`}
                onClick={() => this.onClick(category)}
              >
                {category}
              </Button>
            );
          })}
        </div>
      </section>
    );
  }
}

export default CategoryPicker;
