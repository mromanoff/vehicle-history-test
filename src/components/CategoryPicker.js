import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import { CATEGORIES } from '../constants';

class CategoryPicker extends Component {
  static propTypes = {
    currentCategory: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
  };

  render() {
    return (
      <section>
        <h1 className="title">Pick category</h1>
        <div className="module category-picker">
          {CATEGORIES.map(category => {
            return (
              <Button
                key={category}
                className={`btn btn--tertiary btn--sm ${
                  this.props.currentCategory === category
                    ? 'btn--tertiary-active'
                    : ''
                }`}
                onClick={() => this.props.onChange(category)}
              >
                {category}
              </Button>
            );
          })}
        </div>
      </section>
    );
  }
}

export default CategoryPicker;
