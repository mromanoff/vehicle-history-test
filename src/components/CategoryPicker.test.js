import React from 'react';
import renderer from 'react-test-renderer';
import CategoryPicker from './CategoryPicker';

it('CategoryPicker renders correctly', () => {
  const component = renderer.create(
    <CategoryPicker currentCategory={'Animals'} onChange={() => {}} />,
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onChange('Animals');
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
