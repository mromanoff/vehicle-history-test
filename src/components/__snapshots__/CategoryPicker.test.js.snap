// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPicker renders correctly 1`] = `
<section>
  <h1
    className="title"
  >
    Pick category
  </h1>
  <ul
    className="list-unstyled list-inline category-picker"
  >
    <li
      className="list-inline-item active"
      onClick={[Function]}
    >
      Animals
    </li>
    <li
      className="list-inline-item "
      onClick={[Function]}
    >
      Transportation
    </li>
    <li
      className="list-inline-item "
      onClick={[Function]}
    >
      Nature
    </li>
    <li
      className="list-inline-item "
      onClick={[Function]}
    >
      Sport
    </li>
  </ul>
</section>
`;
